index.js

```diff
@@ -37,41 +38,170 @@ var EchoClient = function(config){
     }
-    if(!data.source){
-        throw new Error('Missing field data.source');
+
+    if (!queryParams) {
+        throw new Error('Missing Analytics queryParams');
+    }
+
+    var constructQueryStringResponse = this._queryStringParams(queryParams);
+
+    if (constructQueryStringResponse.errors) {
+        this.error('One or more invalid analytics queryParams where supplied: ' + constructQueryStringResponse.errors.join());
+        throw new Error('Invalid Analytics queryParams');
     }
 
     var requestOptions = {
-        url: this.config.echo_endpoint+'/1/events',
+        url: this.config.echo_endpoint + '/1/analytics/' + queryOperator + '?' + constructQueryStringResponse.queryString,
+        method: 'GET',
         headers: {
-            'Accept': 'application/json',
-            'Authorization':'Bearer '+token
-        },
-        body:data,
-        method:'POST',
-        json:true
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer ' + token
+        }
     };
 
-    request.post(requestOptions, function(err, response, body){
-        if(err){
+    if (useCache === false) {
+        requestOptions.headers['cache-control'] = 'none';
+    }
+
+    this.debug(JSON.stringify(requestOptions));
+
+    var _this = this;
+
+    request.get(requestOptions, function(err, response, body) {
+        if (err) {
+            _this.error('[echoClient] queryAnalytics error: ' + JSON.stringify(err));
             callback(err);
-        } else{
+        } else {
             callback(null, body);
         }
     });
+};
 
-    this.debug(JSON.stringify(requestOptions));
+/**
+ * Build up a query string
+ * @param {object} params
+ * @returns {string}
+ * @private
+ */
+EchoClient.prototype._queryStringParams = function(params) {
+    var queryString = '';
+    var queryStringParams = [];
+    var paramErrors = null;
+
+    var isValidParameter = function (parameter) {
+        var validParameters = [
+            'class',
+            'source',
+            'property',
+            'interval',
+            'group_by',
+            'key',
+            'value',
+            'from',
+            'to',
+            'percentile',
+            'user',
+            'filter',
+            'n'
+        ];
+
+        var hasDot = parameter.indexOf('.');
 
+        if (hasDot > -1) {
+            parameter = parameter.substring(0, hasDot);
+        }
+
+        if (validParameters.indexOf(parameter) === -1) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    if (!_.isEmpty(params)) {
+        for (var param in params) {
+            if (params.hasOwnProperty(param)) {
+                if (isValidParameter(param)) {
+                    queryStringParams.push(encodeURIComponent(param) + '=' + encodeURIComponent(params[param]));
+                } else {
+                    if (!paramErrors) {
+                        paramErrors = [];
+                    }
```

Comment:
I don't like this. Set it as an empty array and then the caller can check for empty, just checking for a falsy value is lazy.